ucspi-tcp-x/dotls.c

```diff
@@ -0,0 +1,554 @@
+/*
+ * $Log: $
+ */
+#ifdef TLS
+#include <unistd.h>
+#ifdef DARWIN
+#define opteof -1
+#else
+#include <sgetopt.h>
+#endif
+#include <openssl/ssl.h>
+#include <stralloc.h>
+#include <scan.h>
+#include <fmt.h>
+#include <substdio.h>
+#include <strerr.h>
+#include <case.h>
+#include <env.h>
+#include <fd.h>
+#include <sig.h>
+#include <str.h>
+#include <alloc.h>
+#include <gen_alloc.h>
+#include <gen_allocdefs.h>
+#include <timeoutwrite.h>
+#include "pathexec.h"
+#include "tls.h"
+
+#define FATAL         "sslerator: fatal: "
+#define HUGESMTPTEXT  5000
+
+#ifndef	lint
+static char     sccsid[] = "$Id: $";
+#endif
+
+struct substdio ssin, ssto, smtpin, smtpto;
+unsigned long   dtimeout = 60;
+stralloc        certfile = { 0 };
```

Comment:
global variables will be 0-initialized by default anyway.

Reply:
This is a bad habit I have of initializing variables :).